src/Adapter/Cart/CommandHandler/UpdateCartDeliverySettingsHandler.php

```diff
@@ -139,4 +117,89 @@ private function createCartRule(Cart $cart, string $backOfficeOrderCode): CartRu
 
         return $freeShippingCartRule;
     }
+
+    /**
+     * @param Cart $cart
+     * @param UpdateCartDeliverySettingsCommand $command
+     *
+     * @throws CannotDeleteCartRuleException
+     * @throws CartRuleException
+     * @throws PrestaShopException
+     */
+    protected function handleFreeShippingOption(Cart $cart, UpdateCartDeliverySettingsCommand $command): void
+    {
+        $backOfficeOrderCode = sprintf('%s%s', CartRule::BO_ORDER_CODE_PREFIX, $cart->id);
+
+        $freeShippingCartRule = $this->getCartRuleForBackOfficeFreeShipping($backOfficeOrderCode);
+
+        if ($command->allowFreeShipping()) {
+            if (null === $freeShippingCartRule) {
+                $freeShippingCartRule = $this->createCartRule($cart, $backOfficeOrderCode);
+            }
+            $cart->addCartRule((int) $freeShippingCartRule->id);
+
+            return;
+        }
+
+        if (null === $freeShippingCartRule) {
+            return;
+        }
+
+        $cart->removeCartRule((int) $freeShippingCartRule->id);
+
+        try {
+            if (false === $freeShippingCartRule->delete()) {
+                throw new CannotDeleteCartRuleException(sprintf('Failed deleting cart rule #%s', $freeShippingCartRule->id));
+            }
+        } catch (PrestaShopException $e) {
+            throw new CartRuleException(sprintf('An error occurred when trying to delete cart rule #%s', $freeShippingCartRule->id));
+        }
+    }
+
+    /**
+     * @param Cart $cart
+     * @param UpdateCartDeliverySettingsCommand $command
+     *
+     * @throws CartException
+     * @throws PrestaShopException
+     */
+    private function handleGiftOption(Cart $cart, UpdateCartDeliverySettingsCommand $command): void
+    {
+        if ($command->isAGift() !== null) {
```

Comment:
Ideally you should check if the configuration is enabled here as well (for recycle as well), since the configuration service is already injected it's not too much to add either

Reply:
I was hesitating about it, fifty/fifty because I'm not sure it's the domain responsibility to discard part of the Command content because configuration says it's incorrect. Or maybe Command should be rejected because configuration is wrong 🤔

Reply:
I agree with @matks It's not a domain requirement. It should handle what have in its context :thinking:

Reply:
Well the configuration is part of the domain, it enables/disables feature, it changes behaviour sometimes
The handler is supposed to be the "guardian" of the domain, in this case if the feature is disabled it should either ignore it or throw an exception because it's not possible to have a Gift order if the GIft feature is disabled

But if you simply apply it without caring about the configuration it means anyone could use the command however he wants (even directly via the API) regardless of the configuration

It's like the `Return product` configuration is disabled, but I can still use the API to force some refunds without any checking

Reply:
Please have a [octogone](https://www.jewanda.com/2019/03/chronique-mais-cest-quoi-loctogone-de-booba-et-kaaris-en-fait/) and then tell me who is the winner, I'll apply his feedback